js/src/blocks/product-date-time-field/edit.js

```diff
@@ -21,11 +22,18 @@ import styles from './editor.module.scss';
  * @typedef {import('../types.js').ProductBasicAttributes} ProductBasicAttributes
  */
 
-async function resolveValidationMessage( inputRef ) {
+async function resolveValidationMessage( inputRef, context ) {
 	const input = inputRef.current;
 
 	if ( ! input.validity.valid ) {
-		return input.validationMessage;
+		if ( isWcVersion( '9.2.0', '>' ) ) {
```

Comment:
Looks like the comparison should be `<`, and suggest adding a `compatibility-code` comment.

```suggestion
		// compatibility-code "WC < 9.2"
		if ( isWcVersion( '9.2.0', '<' ) ) {
```

Otherwise, with WC < 9.2, it will cause an error after entering an invalid value.

https://github.com/user-attachments/assets/6adddec1-0479-423c-8db7-4ffdd69cd9b7